### DISCUSSION TITLE
Support http as well as https for Html5 exports.

### DISCUSSION THREAD

**Discussion by Nukley** [godotengine/godot-proposals#10075] [General Discussions]
I am sort of curious as to why this isn't already done, as it seems in Firefox the wasm does load.
Exported by Godot 4.3 beta 2 with Thread Support turned off,
See here: http://content.nukley.com/misc/2-test/

Do note I couldn't get this loaded up in konquerer, which is a chromium based browser, it throws this and then hangs:
`Uncaught (in promise) CompileError: WebAssembly.instantiateStreaming(): unexpected section <Exception> @+104383`

The only thing required to make this load in Firefox was the removal of three lines in the error handling section of the main .js file.
Specifically these:
```
if (!Features.isSecureContext()) {
    missing.push('Secure Context - Check web server configuration (use HTTPS)');
}
```
The audio does fail to load the driver, even if a button is specifically added to load an AudioContext.
See here: http://content.nukley.com/misc/1/

What continues to confuse me is godot 3 was capable of doing this fine.
See here (Click the background): http://content.nukley.com/misc/g3daudio/

**Comment by Calinou** [comment]
Web exports that have threads disabled should be able to work in plain HTTP, but this check should remain in web exports with threads enabled.

If there's a way to conditionally disable/remove checks when exporting with threads disabled, then we can modify the default HTML shell accordingly.

**Reply by Nukley** [reply]
Heh, I wasn't expecting a response so I turned this into an issue... Should've known better. Yeah if you can figure this one out that would be cool.

**Reply by Nukley** [reply]
As far as I'm aware I am using it with threads disabled... but I can check again.

**Reply by Nukley** [reply]
Yeah, I've got two examples,
With threads: http://content.nukley.com/misc/withthread/
Without threads: http://content.nukley.com/misc/withoutthread/

http exporting just doesn't work by default, and is heavily broken when bypassed.

**Comment by Calinou** [comment]
- Closing in favor of https://github.com/godotengine/godot-proposals/issues/10076.
